project-koku/koku
# Multi stage container build build

## Jira Ticket

[COST-####](https://issues.redhat.com/browse/COST-####)

## Description

Changes the containerfile to a multi-stage build doing different things based on the image architecture. This transparently builds ARM or x86_64 images.

## Testing

1. Checkout Branch
2. Remove `compose_file` from the `.env` file
3. Run `docker compose up -d --build`
4. Verify image builds successfully and matches the host architecture: `docker image inspect -f '{{ .RepoTags }} {{ .Architecture }} {{ .Size }}' koku_base`

## Notes

Using native images allows `watchdog` to work properly but only when using Docker Desktop since it uses `grpcfuse` for bind mounts. I tested used 9p with Rancher Desktop, but inotify events from the container host don't make it processes running in the container.
